contents/overlay/context-menus/mail-box/index.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  Center,
+  Text,
+  Tabs,
+  Tab,
+  TabPanel,
+  ContextMenu,
+  ContextMenuTrigger,
+  MenuList,
+  MenuItem,
+  HStack,
+} from "@yamada-ui/react"
+import { type Dispatch, type SetStateAction, type FC, useState } from "react"
+
+const mailsData = [
+  {
+    id: 1,
+    title: "GitHub",
+    description: "You assigned on review. #905",
+    time: "19:20",
+  },
+  {
+    id: 2,
+    title: "GitHub",
+    description: "You assigned on review. #904",
+    time: "19:13",
+  },
+  {
+    id: 3,
+    title: "GitHub",
+    description: "You assigned on review. #903",
+    time: "19:10",
+  },
+  {
+    id: 4,
+    title: "GitHub",
+    description: "You assigned on review. #902",
+    time: "19:00",
+  },
+  {
+    id: 5,
+    title: "GitHub",
+    description: "You assigned on review. #901",
+    time: "18:55",
+  },
+]
+
+const MailItem: FC<{
+  id: number
+  currentId: number
+  title: string
+  description: string
+  time: string
+  isArchived?: boolean
+  handleDelete?: () => void
+  handleArchive?: () => void
+  setCurrentId: Dispatch<SetStateAction<number>>
+}> = ({
+  id,
+  currentId,
+  time,
+  description,
+  title,
+  isArchived,
+  handleArchive,
+  handleDelete,
+  setCurrentId,
+}) => {
+  return (
+    <ContextMenu
+      onOpen={() => {
+        setCurrentId(id)
+      }}
+    >
+      <ContextMenuTrigger
+        as={Center}
+        w="full"
+        p="md"
+        flexDir="row"
+        justifyContent="space-between"
+        borderBottomWidth="1px"
+        borderBottomStyle="solid"
+        bgColor={id === currentId ? "primary" : undefined}
+      >
+        <HStack gap="md">
+          <Text>{title}</Text>
+          <Text>{description}</Text>
+        </HStack>
+        <Center>
+          <Text>{time}</Text>
+        </Center>
+      </ContextMenuTrigger>
+
+      <MenuList>
+        <MenuItem onClick={handleDelete}>Delete</MenuItem>
+        <MenuItem onClick={handleArchive}>
+          {isArchived ? "Remove Archive" : "Archive"}
+        </MenuItem>
+      </MenuList>
```

Comment:
Maybe you could add icon to menu items?
```suggestion
      <MenuList>
        <MenuItem icon={<TrashIcon color="danger" />} onClick={handleDelete}>
          Delete
        </MenuItem>
        <MenuItem
          icon={
            isArchived ? (
              <ArchiveRestoreIcon color="success" />
            ) : (
              <ArchiveIcon color="danger" />
            )
          }
          onClick={handleArchive}
        >
          {isArchived ? "Remove Archive" : "Archive"}
        </MenuItem>
      </MenuList>
```

Reply:
I accept you suggestion.